src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -213,13 +200,15 @@ public static void RuntimeFailFast(RhFailFastReason reason, Exception? exception
         }
 
         [DoesNotReturn]
-        internal static void FailFast(string message, Exception? exception, RhFailFastReason reason, IntPtr pExAddress, IntPtr pExContext)
+        internal static unsafe void FailFast(string message, Exception? exception, RhFailFastReason reason, IntPtr pExAddress, IntPtr pExContext)
         {
             // If this a recursive call to FailFast, avoid all unnecessary and complex activity the second time around to avoid the recursion
             // that got us here the first time (Some judgement is required as to what activity is "unnecessary and complex".)
             bool minimalFailFast = InFailFast.Value || (exception == PreallocatedOutOfMemoryException.Instance);
             InFailFast.Value = true;
 
+            _crashInfo.Open(reason, message);
```

Comment:
I think that multiple threads can be running this in parallel (it is fairly common to see situations where multiple threads crash at about the same time).

I do not see any synchronization that would prevent all of them from using the same instance.

Reply:
Here is a small test that I have used to see what it does today:
```csharp
class Program
{
    static volatile bool go = false;

    static void Main()
    {
        for (int i = 0; i < 5; i++)
            new Thread(Crash).Start();
         Thread.Sleep(10);

         go = true;
         Crash();
    }

    static void Crash() => Crash(Random.Shared.Next(20));

    static void Crash(int depth)
    {
        if (depth > 0) Crash(depth - 1);

        while (!go) { };

        throw new Exception();
    }
}
```